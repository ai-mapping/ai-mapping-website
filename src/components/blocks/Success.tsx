import React from 'react';
import SuccessCard from './SuccessCard';
import Section from '@/components/layout/Section';

// Define types for case studies
export type CaseStudy = {
  id: string;
  title: string;
  deliveryTime: string;
  description: string;
  image: {
    src: string;
    alt: string;
    width: number;
    height: number;
  };
};

// Main props interface
export type SuccessProps = {
  heading: string;
  subheading: string;
  cases: CaseStudy[];
  buttonText: string;
};

// Default props
export const defaultSuccessProps: SuccessProps = {
  heading: 'Примеры успешных поставок',
  subheading: 'Реальные кейсы доставки запчастей нашим клиентам',
  cases: [
    {
      id: '1',
      title: 'Гидроцилиндр для CAT 320D',
      deliveryTime: 'Доставлено за 9 дней',
      description: 'Клиент из Подмосковья искал гидроцилиндр для экскаватора CAT 320D. Официальные дилеры называли срок от 2-х месяцев.',
      image: {
        src: 'https://via.placeholder.com/600x400',
        alt: 'Гидроцилиндр CAT 320D',
        width: 400,
        height: 192
      }
    },
    {
      id: '2',
      title: 'Турбина для KOMATSU PC300',
      deliveryTime: 'Доставлено за 12 дней',
      description: 'Строительная компания из Москвы срочно нуждалась в турбине для экскаватора, который простаивал более месяца.',
      image: {
        src: 'https://via.placeholder.com/600x400',
        alt: 'Турбина KOMATSU PC300',
        width: 400,
        height: 192
      }
    },
    {
      id: '3',
      title: 'Блок управления CAT D6R',
      deliveryTime: 'Доставлено за 11 дней',
      description: 'Редкий электронный блок управления для бульдозера CAT D6R. Клиент не мог найти его у других поставщиков.',
      image: {
        src: 'https://via.placeholder.com/600x400',
        alt: 'Блок управления CAT D6R',
        width: 400,
        height: 192
      }
    }
  ],
  buttonText: 'Заказать такой же'
};

const Success = ({
  heading = defaultSuccessProps.heading,
  subheading = defaultSuccessProps.subheading,
  cases = defaultSuccessProps.cases,
  buttonText = defaultSuccessProps.buttonText
}: Partial<SuccessProps>) => {
  return (
    <Section background="background" paddingY="lg" containerWidth="container">
        <h2 className="text-2xl md:text-3xl font-semibold text-center mb-3 text-white">
          {heading}
        </h2>
        <p className="text-neutral text-center mb-10 max-w-2xl mx-auto">
          {subheading}
        </p>

        <div className="grid grid-cols-1 md:grid-cols-3 gap-8">
          {cases.map((caseStudy) => (
            <SuccessCard
              key={caseStudy.id}
              caseStudy={caseStudy}
              buttonText={buttonText}
            />
          ))}
        </div>
    </Section>
  );
};

export default Success;
