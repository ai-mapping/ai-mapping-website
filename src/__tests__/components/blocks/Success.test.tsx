import React from 'react';
import { render, screen } from '@testing-library/react';
import Success, { defaultSuccessProps } from '@/components/blocks/Success';

// Mock the SuccessCard component
jest.mock('@/components/blocks/SuccessCard', () => {
  return {
    __esModule: true,
    default: ({ caseStudy, buttonText }) => (
      <div data-testid={`success-card-${caseStudy.id}`}>
        <h3>{caseStudy.title}</h3>
        <div>{caseStudy.deliveryTime}</div>
        <p>{caseStudy.description}</p>
        {/* eslint-disable-next-line @next/next/no-img-element */}
        <img 
          src={caseStudy.image.src} 
          alt={caseStudy.image.alt} 
          width={caseStudy.image.width} 
          height={caseStudy.image.height} 
        />
        <button>{buttonText}</button>
      </div>
    )
  };
});

describe('Success Component', () => {
  it('renders with default props correctly', () => {
    const { container } = render(<Success />);
    
    // Check if heading and subheading are rendered
    expect(screen.getByText(defaultSuccessProps.heading)).toBeInTheDocument();
    expect(screen.getByText(defaultSuccessProps.subheading)).toBeInTheDocument();
    
    // Check if all case studies are rendered as cards
    defaultSuccessProps.cases.forEach((caseStudy) => {
      const card = screen.getByTestId(`success-card-${caseStudy.id}`);
      expect(card).toBeInTheDocument();
      expect(screen.getByText(caseStudy.title)).toBeInTheDocument();
      expect(screen.getByText(caseStudy.deliveryTime)).toBeInTheDocument();
      expect(screen.getByText(caseStudy.description)).toBeInTheDocument();
    });
    
    // Check if button text is rendered
    const buttons = screen.getAllByText(defaultSuccessProps.buttonText);
    expect(buttons.length).toBe(defaultSuccessProps.cases.length);
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
  
  it('renders with custom props correctly', () => {
    const customProps = {
      heading: 'Custom Success Heading',
      subheading: 'Custom Success Subheading',
      cases: [
        {
          id: 'custom-1',
          title: 'Custom Case Study',
          deliveryTime: 'Custom Delivery Time',
          description: 'Custom Description',
          image: {
            src: '/custom-image.jpg',
            alt: 'Custom Image Alt',
            width: 300,
            height: 150
          }
        }
      ],
      buttonText: 'Custom Button Text'
    };
    
    const { container } = render(<Success {...customProps} />);
    
    // Check if custom heading and subheading are rendered
    expect(screen.getByText(customProps.heading)).toBeInTheDocument();
    expect(screen.getByText(customProps.subheading)).toBeInTheDocument();
    
    // Check if custom case study is rendered
    const card = screen.getByTestId(`success-card-${customProps.cases[0].id}`);
    expect(card).toBeInTheDocument();
    expect(screen.getByText(customProps.cases[0].title)).toBeInTheDocument();
    expect(screen.getByText(customProps.cases[0].deliveryTime)).toBeInTheDocument();
    expect(screen.getByText(customProps.cases[0].description)).toBeInTheDocument();
    
    // Check if custom button text is rendered
    expect(screen.getByText(customProps.buttonText)).toBeInTheDocument();
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});