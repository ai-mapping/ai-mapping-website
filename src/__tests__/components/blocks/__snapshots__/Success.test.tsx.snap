// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success Component renders with custom props correctly 1`] = `
<div>
  <section
    class="bg-background py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8"
    >
      <h2
        class="text-2xl md:text-3xl font-semibold text-center mb-3 text-white"
      >
        Custom Success Heading
      </h2>
      <p
        class="text-neutral text-center mb-10 max-w-2xl mx-auto"
      >
        Custom Success Subheading
      </p>
      <div
        class="grid grid-cols-1 md:grid-cols-3 gap-8"
      >
        <div
          data-testid="success-card-custom-1"
        >
          <h3>
            Custom Case Study
          </h3>
          <div>
            Custom Delivery Time
          </div>
          <p>
            Custom Description
          </p>
          <img
            alt="Custom Image Alt"
            height="150"
            src="/custom-image.jpg"
            width="300"
          />
          <button>
            Custom Button Text
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Success Component renders with default props correctly 1`] = `
<div>
  <section
    class="bg-background py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8"
    >
      <h2
        class="text-2xl md:text-3xl font-semibold text-center mb-3 text-white"
      >
        Примеры успешных поставок
      </h2>
      <p
        class="text-neutral text-center mb-10 max-w-2xl mx-auto"
      >
        Реальные кейсы доставки запчастей нашим клиентам
      </p>
      <div
        class="grid grid-cols-1 md:grid-cols-3 gap-8"
      >
        <div
          data-testid="success-card-1"
        >
          <h3>
            Гидроцилиндр для CAT 320D
          </h3>
          <div>
            Доставлено за 9 дней
          </div>
          <p>
            Клиент из Подмосковья искал гидроцилиндр для экскаватора CAT 320D. Официальные дилеры называли срок от 2-х месяцев.
          </p>
          <img
            alt="Гидроцилиндр CAT 320D"
            height="192"
            src="https://via.placeholder.com/600x400"
            width="400"
          />
          <button>
            Заказать такой же
          </button>
        </div>
        <div
          data-testid="success-card-2"
        >
          <h3>
            Турбина для KOMATSU PC300
          </h3>
          <div>
            Доставлено за 12 дней
          </div>
          <p>
            Строительная компания из Москвы срочно нуждалась в турбине для экскаватора, который простаивал более месяца.
          </p>
          <img
            alt="Турбина KOMATSU PC300"
            height="192"
            src="https://via.placeholder.com/600x400"
            width="400"
          />
          <button>
            Заказать такой же
          </button>
        </div>
        <div
          data-testid="success-card-3"
        >
          <h3>
            Блок управления CAT D6R
          </h3>
          <div>
            Доставлено за 11 дней
          </div>
          <p>
            Редкий электронный блок управления для бульдозера CAT D6R. Клиент не мог найти его у других поставщиков.
          </p>
          <img
            alt="Блок управления CAT D6R"
            height="192"
            src="https://via.placeholder.com/600x400"
            width="400"
          />
          <button>
            Заказать такой же
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
